micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -692,7 +695,12 @@ private boolean isStaleId(Id originalId) {
      * @return {@code true} if the id is stale
      */
     private boolean unmarkStaleId(Id originalId) {
-        return !stalePreFilterIds.isEmpty() && stalePreFilterIds.remove(originalId);
+        if (!stalePreFilterIds.isEmpty()) {
+            synchronized (meterMapLock) {
```

Comment:
This isn't needed for fixing the CME and it's not clear to me what scenario this improves (it also doesn't seem to hurt anything either, though). I am leaning toward not including this change unless we have a concrete idea on that.

Reply:
@shakuzen You're right it's not needed for fixing the actual issue, but it was mainly provided after @lenin-jaganathan comment that all writes(write ops) should be covered as we also stated in the comment of the collection. 

But I'll update it.